pyathena/common.py

```diff
@@ -142,24 +142,53 @@ def _build_start_query_execution_request(self, query, work_group=None, s3_stagin
             })
         return request
 
-    def _execute(self, operation, parameters=None, work_group=None, s3_staging_dir=None):
+    def _find_previous_query_id(self, request, work_group, cache_size):
+        next_token = None
+        while cache_size > 0:
+            n = min(cache_size, 50)  # 50 is max allowed by AWS API
+            cache_size -= n
+            request_params = {'MaxResults': n}
+            if work_group is not None:
+                request_params['WorkGroup'] = work_group
+            if next_token is not None:
+                request_params['NextToken'] = next_token
+            response = self.connection._client.list_query_executions(**request_params)
+            query_ids = response['QueryExecutionIds']
+            next_token = response['NextToken']
```

Comment:
If the cache size is set to 1000 or larger, a large number of requests will be thrown even if there is almost no query execution history.
This loop should be stopped when the next_token of the response from the API is None.